main_test.go

```diff
@@ -0,0 +1,1986 @@
+			"U%[5]d:x:%[5]d:%[6]d::/:/bin/sh\n"+
+			"U%[7]d:x:%[7]d:%[8]d::/:/bin/sh\n"+
+			"U%[9]d:x:%[9]d:%[10]d::/:/bin/sh\n"+
+			"U%[11]d:x:%[11]d:%[12]d::/:/bin/sh\n",
+			u.Uid, u.Gid, UserA, GroupA, UserB, GroupB, UserC, GroupC, UserD, GroupD, UserE, GroupE))
+
+		writeFileString(t, groups, fmt.Sprintf(""+
+			"root:x:0:\n"+
+			"group:x:%[1]s:user::/:/bin/sh\n"+
+			"G%[2]d:x:%[2]d:U%[3]d::/:/bin/sh\n"+
+			"G%[4]d:x:%[4]d:U%[5]d::/:/bin/sh\n"+
+			"G%[6]d:x:%[6]d:U%[7]d::/:/bin/sh\n"+
+			"G%[8]d:x:%[8]d:U%[9]d::/:/bin/sh\n"+
+			"G%[10]d:x:%[10]d:U%[11]d::/:/bin/sh\n",
+			u.Gid, GroupA, UserA, GroupB, UserB, GroupC, UserC, GroupD, UserD, GroupE, UserE))
+
+		cmd := exec.Command("singularity", "run", //nolint:gosec
+			"--bind", tmpTemp+":/tmp,"+users+":/etc/passwd,"+groups+":/etc/group,"+
+				binDir+":/build,"+runScript+":/run.sh",
+			"--home", tmpHome,
+			"--overlay", files, sif, "/run.sh")
+
+		So(cmd.Run(), ShouldBeNil)
+
+		userBaseDirs := fmt.Sprintf(``+
+			"U%[1]d\t\t/objects/store1/data/sheets\t%[6]d\t10240\t0\t2\t0\tOK\n"+
+			"U%[1]d\t\t/objects/store2/part0/teams/team1\t%[6]d\t300\t0\t2\t0\tOK\n"+
+			"U%[1]d\t\t/simple/A\t%[6]d\t1\t0\t1\t0\tOK\n"+
+			"U%[2]d\t\t/objects/store1/data/dbs\t%[6]d\t66666\t0\t2\t0\tOK\n"+
+			"U%[2]d\t\t/objects/store2/important/docs\t%[6]d\t1200\t0\t1\t0\tOK\n"+
+			"U%[2]d\t\t/objects/store2/part0/teams/team2\t%[6]d\t1000\t0\t1\t0\tOK\n"+
+			"U%[3]d\t\t/objects/store1/data/temp\t%[6]d\t6000\t0\t3\t0\tOK\n"+
+			"U%[4]d\t\t/objects/store2/part1/other/myDir\t%[6]d\t2048\t0\t1\t0\tOK\n"+
+			"U%[5]d\t\t/simple/E\t%[6]d\t2\t0\t1\t0\tOK", UserA, UserB, UserC, UserD, UserE,
+			time.Now().Unix()/86400,
+		)
+
+		ub, err := fs.Glob(os.DirFS(tmpTemp), filepath.Join("final", "*basedirs.userusage.tsv"))
+		So(err, ShouldBeNil)
+		So(len(ub), ShouldEqual, 1)
+
+		compareFileContents(t, filepath.Join(tmpTemp, ub[0]), userBaseDirs)
+
+		groupBaseDirs := fmt.Sprintf(``+
+			"G%[1]d\t\t/objects/store1/data/dbs\t%[5]d\t66666\t0\t2\t0\tNot OK\n"+
+			"G%[1]d\t\t/objects/store1/data/sheets\t%[5]d\t10240\t0\t2\t0\tNot OK\n"+
+			"G%[1]d\t\t/objects/store1/data/temp\t%[5]d\t6000\t0\t3\t0\tNot OK\n"+
+			"G%[1]d\t\t/objects/store2/part0/teams/team1\t%[5]d\t100\t0\t1\t0\tNot OK\n"+
+			"G%[1]d\t\t/objects/store2/part1/other/myDir\t%[5]d\t2048\t0\t1\t0\tNot OK\n"+
+			"G%[1]d\t\t/simple/A\t%[5]d\t1\t0\t1\t0\tNot OK\n"+
+			"G%[2]d\t\t/objects/store2/part0/teams/team1\t%[5]d\t200\t0\t1\t0\tNot OK\n"+
+			"G%[2]d\t\t/objects/store2/part0/teams/team2\t%[5]d\t1000\t0\t1\t0\tNot OK\n"+
+			"G%[3]d\t\t/objects/store2/important/docs\t%[5]d\t1200\t0\t1\t0\tNot OK\n"+
+			"G%[4]d\t\t/simple/E\t%[5]d\t2\t0\t1\t0\tNot OK",
+			GroupA, GroupB, GroupD, GroupE,
+			time.Now().Unix()/86400,
+		)
+
+		gb, err := fs.Glob(os.DirFS(tmpTemp), filepath.Join("final", "*basedirs.groupusage.tsv"))
+		So(err, ShouldBeNil)
+		So(len(gb), ShouldEqual, 1)
+
+		compareFileContents(t, filepath.Join(tmpTemp, gb[0]), groupBaseDirs)
```

Comment:
Did you not want to test all the output file types in this PR, including doing a test that the resulting databases work with a test of `wrstat where`?